File: find_headers.js
const fs = require('fs');
const path = require('path');

function getFiles(dir, files = []) {
  const fileList = fs.readdirSync(dir);
  for (const file of fileList) {
    const name = `${dir}/${file}`;
    if (fs.statSync(name).isDirectory()) {
      getFiles(name, files);
    } else {
      if (name.endsWith('.tsx')) {
        files.push(name);
      }
    }
  }
  return files;
}

const files = getFiles('./client/src/pages');
const results = [];

for (const file of files) {
  const content = fs.readFileSync(file, 'utf-8');
  const regex = /<PageHeader[^>]*title=["']([^"']+)["']/g;
  let match;
  while ((match = regex.exec(content)) !== null) {
    results.push({ file, original: match[1] });
  }
}

console.log(JSON.stringify(results, null, 2));

File: find_headers.py
import os
import re
import json

results = []

def get_files(directory):
    for root, dirs, files in os.walk(directory):
        for file in files:
            if file.endswith('.tsx'):
                yield os.path.join(root, file)

for filepath in get_files('./client/src/pages'):
    try:
        with open(filepath, 'r', encoding='utf-8') as f:
            content = f.read()
            matches = re.finditer(r'<PageHeader[^>]*title=["\']([^"\']+)["\']', content)
            for match in matches:
                results.append({
                    "file": filepath,
                    "original": match.group(1)
                })
    except Exception as e:
        pass

print(json.dumps(results, indent=2))
